app/src/main/java/io/rapidpro/androidchannel/IncomingSMSObserver.java

```diff
@@ -61,8 +64,18 @@ public void onChange(boolean selfChange) {
 
         Uri inboxUri = Uri.parse("content://sms/inbox");
 
+        String ignoredMessage = prefs.getString(SettingsActivity.SMS_IGNORED_MESSAGE, "");
+        String selection = null;
+        String [] selectionArgs = null;
+
+        if (!TextUtils.isEmpty(ignoredMessage)) {
+            selection = "body NOT LIKE ?";
+            selectionArgs = new String[]{"%" + ignoredMessage + "%"};
```

Comment:
This seems like prone to SQL injection attacks (though it matters less in this case than a server). Might just be easier to do a substring match when we are actually doing the iteration.